Source/WebCore/animation/AnimationEffect.cpp

```diff
@@ -228,8 +228,10 @@ ExceptionOr<void> AnimationEffect::updateTiming(Document& document, std::optiona
     if (timing->iterations)
         m_timing.iterations = timing->iterations.value();
 
-    if (timing->duration)
-        m_timing.iterationDuration = std::holds_alternative<double>(timing->duration.value()) ? Seconds::fromMilliseconds(std::get<double>(timing->duration.value())) : 0_s;
+    if (auto duration = timing->duration) {
+        m_hasAutoDuration = std::holds_alternative<String>(*duration);
```

Comment:
Probably someone else should review this, but this seemed a little odd to me. Previously it was assigned a time, but now a boolean?

Reply:
We're setting the new `m_hasAutoDuration` flag here, which is indeed a `bool`. We now set `m_timing.iterationDuration` under the new `normalizeSpecifiedTiming()` method.